### PR TITLE
Chapter 7 - Sign Up

### PR DESCRIPTION
**Pull request by EvonneNgXinYi**
Chapter 7 - Sign Up

### CODE REVIEW

**Comment by EvonneNgXinYi** [comment]
ready

File: app/helpers/users_helper.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+module UsersHelper
+  def gravatar_for user
+    gravatar_id = Digest::MD5::hexdigest(user.email.downcase)
```
Comment (chungpv-1008) [inline app/helpers/users_helper.rb]:
```suggestion
    gravatar_id = Digest::MD5::hexdigest user.email.downcase
```
please add end line (blank line) for this file

File: app/views/layouts/_shim.html.erb [hunk 2]
```diff
@@ -0,0 +1,4 @@
+<!--[if lt IE 9]>
+  <script src="//cdnjs.cloudflare.com/ajax/libs/html5shiv/r29/html5.min.js">
+  </script>
+<![endif]-->
```
Comment (chungpv-1008) [inline app/views/layouts/_shim.html.erb:4]: please remove this file because not used yet

File: app/views/layouts/application.html.erb [hunk 3]
```diff
@@ -7,13 +7,19 @@
...
 
   <body>
     <%= render "layouts/header" %>
     <div class="container">
+      <% flash.each do |message_type, message| %>
+        <div class="alert alert-<%= message_type %>"><%= message %></div>
+      <% end %>
       <%= yield %>
       <%= render "layouts/footer" %>
+      <%= debug(params) if Rails.env.development? %>
```
Comment (chungpv-1008) [inline app/views/layouts/application.html.erb]:
```suggestion
      <%= debug params if Rails.env.development? %>
```

File: app/views/shared/_error_messages.html.erb [hunk 4]
```diff
@@ -0,0 +1,13 @@
+<% if @user.errors.any? %>
+  <div id="error_explanation">
+    <div class="alert alert-danger">
+       The form contains <%= pluralize(@user.errors.count, "error") %>.
```
Comment (chungpv-1008) [inline app/views/shared/_error_messages.html.erb]:
```suggestion
       The form contains <%= pluralize @user.errors.size, "error" %>.
```

File: config/locales/en.yml [hunk 5]
```diff
@@ -47,7 +47,14 @@ en:
...
     new:
       sign_up_title: "Sign Up"
       heading: "Sign Up"
       description_title: "This will be a sign up page for new users."
+      sign_up_create: "Create my account"
+      confirmation: "Confirmation"
+    create:
+      success_mess_create: "Welcome to Sample App!"
+      fail_mess_create: "Account created fail !!!"
+
```
Comment (chungpv-1008) [inline config/locales/en.yml]: add end line

File: config/routes.rb [hunk 6]
```diff
@@ -4,6 +4,7 @@
     get "/help", to: "static_pages#help"
     get "/about", to: "static_pages#about"
     get "/signup", to: "users#new"
-    get "users/new"
+    get "sessions/new"
```
Comment (chungpv-1008) [inline config/routes.rb]:
```suggestion
    get "/login", to: "sessions#new"
```

**Comment by EvonneNgXinYi** [comment]
ready

File: app/views/layouts/_shim.html.erb (hunk 2 above)
Comment (EvonneNgXinYi) [reply app/views/layouts/_shim.html.erb:4]: not used yet? From the book, the shim file only be taught in Chap 5. However，i missed it before.

File: app/assets/stylesheets/variables.scss [hunk 7]
```diff
@@ -3,3 +3,6 @@ $dark-gray: #222;
 $gray: #555;
 $light-gray: #777;
 $white: #fff;
+$silver: #bbb;
+$red: #e02b11;
+$state-danger-text: b94a48;
```
Comment (saobv-2223) [inline app/assets/stylesheets/variables.scss]: add #

File: app/helpers/users_helper.rb [hunk 8]
```diff
@@ -0,0 +1,8 @@
+module UsersHelper
+  def gravatar_for user
+    gravatar_id = Digest::MD5::hexdigest user.email.downcase
+    gravatar_url = "#{Settings.links.gravatar}#{gravatar_id}"
+    image_tag gravatar_url, alt: user.name, class: "gravatar"
+  end
+ end
+
```
Comment (saobv-2223) [inline app/helpers/users_helper.rb]: add endline

File: app/views/layouts/_shim.html.erb (hunk 2 above)
Comment (saobv-2223) [reply app/views/layouts/_shim.html.erb:4]: oke, no problems

**Comment by EvonneNgXinYi** [comment]
ready

**Comment by ngocvt-0484** [comment]
merged

**Pull request merged**